File: backend/src/main/java/com/example/backendapplication/model/Subtask.java
// TODO: Solution File. Remove
package com.example.backendapplication.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.UUID;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "subtasks")
public class Subtask {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.EAGER)
    private Task parentId;

    private String name;

    private String details;

    private boolean completed;

    private Date createdDate;

    private Date updatedDate;
}

File: frontend/src/types/SubTask.ts
// TODO: Define the SubTask type, refer to the back-end code for the shape of a SubTask object.

// Solution
import { Task } from "./Task";

export type SubTask = {
  id: string;
  parentId: Task;
  details?: string;
  completed: boolean;
  createdDate?: string;
  updatedDate?: string;
} & CreateSubTask;

export type CreateSubTask = {
  parentTask: string;
  name: string;
};
